### PR TITLE
feat(nextcloud): #DRIV-19 add new route for moving files from workspa…

### PR DESCRIPTION
**Pull request by aymericdevriesere**
## Describe your changes
Create a new route to move files from workspace to Nextcloud. Nextcloud and Workspace need to belong to the same user.
## Checklist tests
Route tested with Postman
## Issue ticket number and link
[ DRIV-19 ]
https://entsupport.gdapublic.fr/projects/DRIV/issues/DRIV-19
[ PUT ] /files/user/:userid/nextcloud/move
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/nextcloud/controller/DocumentsController.java [hunk 1]
```diff
@@ -182,4 +182,22 @@ public void copyToWorkspace(HttpServerRequest request) {
...
     }
 
+    @Put("/files/user/:userid/nextcloud/move")
+    @ApiDoc("Move a file from ENT workspace to cloud")
+    @SecuredAction(value = "", type = ActionType.RESOURCE)
+    @ResourceFilter(OwnerFilter.class)
+    public void moveToCloud(HttpServerRequest request) {
+        List<String> listFiles = request.params().getAll(Field.ID);
+        String parentId = request.params().get(Field.PARENTNAME);
+        //TODO rename parent id to parent name
```
Comment (BastienBenard) [inline src/main/java/fr/openent/nextcloud/controller/DocumentsController.java:192]: Commentaire

**Comment by JinDouang** [comment]
image

Pense à maj ta branche

**Review by BastienBenard — APPROVED** [review]

File: src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java [hunk 2]
```diff
@@ -42,6 +46,7 @@ public DefaultDocumentsService(ServiceFactory serviceFactory) {
         this.nextcloudConfig = serviceFactory.nextcloudConfig();
         this.storage = serviceFactory.storage();
         this.workspaceHelper = serviceFactory.workspaceHelper();
+        this.eventBus = serviceFactory.eventBus();
```
Comment (JinDouang) [inline src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:49]: utilité ? Je ne vois pas l'usage

File: src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java [hunk 3]
```diff
@@ -34,6 +36,8 @@ public class DefaultDocumentsService implements DocumentsService {
     private final NextcloudConfig nextcloudConfig;
     private final Storage storage;
     private final WorkspaceHelper workspaceHelper;
+    private final EventBus eventBus;
```
Comment (JinDouang) [inline src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:39]: utilité ? Je ne vois pas l'usage

Comment (aymericdevriesere) [reply src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:39]: Vu avec Jin et réglé

File: src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java (hunk 2 above)
Comment (aymericdevriesere) [reply src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:49]: Vu avec Jin et réglé

**Pull request merged**